env/flink/env_flink.cc

```diff
@@ -1,10 +1,844 @@
+  jboolean created = jniEnv->CallBooleanMethod(
+      file_system_instance_, deleteMethod.javaMethod, pathInstance, recursive);
+  jniEnv->DeleteLocalRef(pathInstance);
+
+  status = CurrentStatus([filePath]() {
+    return std::string("Exception when Delete, path: ").append(filePath);
+  });
+  if (!status.ok()) {
+    return status;
+  }
+
+  return created ? IOStatus::OK() : IOStatus::IOError(status.ToString());
+}
+
+IOStatus FlinkFileSystem::CreateDir(const std::string& file_name,
+                                    const IOOptions& options,
+                                    IODebugContext* dbg) {
+  IOStatus s = FileExists(file_name, options, dbg);
+  if (!s.ok()) {
+    return CreateDirIfMissing(file_name, options, dbg);
+  }
+  return IOStatus::IOError(std::string("Exception when CreateDir because Dir (")
+                               .append(file_name)
+                               .append(") exists"));
+}
+
+IOStatus FlinkFileSystem::CreateDirIfMissing(const std::string& file_name,
+                                             const IOOptions& options,
+                                             IODebugContext* dbg) {
+  JNIEnv* jniEnv = getJNIEnv();
+
+  std::string filePath = ConstructPath(file_name);
+  // Construct Path Instance
+  jobject pathInstance;
+  IOStatus status =
+      class_cache_->ConstructPathInstance(filePath, &pathInstance);
+  if (!status.ok()) {
+    return status;
+  }
+
+  // Call mkdirs method
+  JavaClassCache::JavaMethodContext mkdirMethod =
+      class_cache_->GetJMethod(JavaClassCache::JM_FLINK_FILE_SYSTEM_MKDIR);
+  jboolean created = jniEnv->CallBooleanMethod(
+      file_system_instance_, mkdirMethod.javaMethod, pathInstance);
+  jniEnv->DeleteLocalRef(pathInstance);
+  status = CurrentStatus([filePath]() {
+    return std::string("Exception when CreateDirIfMissing, path: ")
+        .append(filePath);
+  });
+  if (!status.ok()) {
+    return status;
+  }
+
+  return created ? IOStatus::OK() : IOStatus::IOError(status.ToString());
+}
+
+IOStatus FlinkFileSystem::GetFileSize(const std::string& file_name,
+                                      const IOOptions& options, uint64_t* size,
+                                      IODebugContext* dbg) {
+  JNIEnv* jniEnv = getJNIEnv();
+  jobject fileStatus;
+  IOStatus status = GetFileStatus(file_name, options, dbg, &fileStatus);
+  if (!status.ok()) {
+    return status;
+  }
+
+  JavaClassCache::JavaMethodContext getLenMethod =
+      class_cache_->GetJMethod(JavaClassCache::JM_FLINK_FILE_STATUS_GET_LEN);
+  jlong fileSize = jniEnv->CallLongMethod(fileStatus, getLenMethod.javaMethod);
+  jniEnv->DeleteLocalRef(fileStatus);
+
+  status = CurrentStatus([file_name]() {
+    return std::string("Exception when GetFileSize, file name: ")
+        .append(file_name);
+  });
+  if (!status.ok()) {
+    return status;
+  }
+
+  *size = fileSize;
+  return IOStatus::OK();
+}
+
+// 1. Must guarantee that FileStatus::getPath will not be used!
```

Comment:
Can you explain why it must be guaranteed that FileStatus::getPath cannot be used here?

Reply:
Thanks for the reminder.
It works well after I reconstruct the logic, so I just removed the comment.